File: android/VelaWallet/app/src/main/java/app/getvela/wallet/ui/components/VelaComponents.kt
package app.getvela.wallet.ui.components

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.Path
import androidx.compose.ui.graphics.drawscope.Fill
import androidx.compose.ui.graphics.drawscope.Stroke
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import app.getvela.wallet.ui.theme.*

// MARK: - Navigation Bar (matches iOS VelaNavBar)

@Composable
fun VelaNavBar(
    title: String,
    onBack: (() -> Unit)? = null,
) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = VelaSpacing.screenH, vertical = 8.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        if (onBack != null) {
            IconButton(onClick = onBack, modifier = Modifier.size(36.dp)) {
                Icon(
                    imageVector = Icons.AutoMirrored.Filled.ArrowBack,
                    contentDescription = "Back",
                    tint = VelaColor.textPrimary,
                )
            }
        } else {
            Spacer(Modifier.size(36.dp))
        }

        Spacer(Modifier.weight(1f))

        Text(
            text = title,
            style = VelaTypography.title(17f),
            color = VelaColor.textPrimary,
        )

        Spacer(Modifier.weight(1f))
        Spacer(Modifier.size(36.dp))
    }
}

// MARK: - Primary Button (matches iOS VelaPrimaryButtonStyle)

@Composable
fun VelaPrimaryButton(
    text: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    enabled: Boolean = true,
    isLoading: Boolean = false,
) {
    Button(
        onClick = onClick,
        modifier = modifier.fillMaxWidth().height(54.dp),
        enabled = enabled && !isLoading,
        shape = VelaButtonShape,
        colors = ButtonDefaults.buttonColors(
            containerColor = VelaColor.textPrimary,
            contentColor = Color.White,
            disabledContainerColor = VelaColor.textPrimary.copy(alpha = 0.5f),
            disabledContentColor = Color.White.copy(alpha = 0.7f),
        ),
    ) {
        if (isLoading) {
            CircularProgressIndicator(
                modifier = Modifier.size(20.dp),
                color = Color.White,
                strokeWidth = 2.dp,
            )
        } else {
            Text(text, style = VelaTypography.label(16f))
        }
    }
}

// MARK: - Secondary Button (matches iOS VelaSecondaryButtonStyle)

@Composable
fun VelaSecondaryButton(
    text: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    enabled: Boolean = true,
) {
    OutlinedButton(
        onClick = onClick,
        modifier = modifier.fillMaxWidth().height(54.dp),
        enabled = enabled,
        shape = VelaButtonShape,
        border = BorderStroke(1.5.dp, VelaColor.border),
        colors = ButtonDefaults.outlinedButtonColors(
            contentColor = VelaColor.textPrimary,
        ),
    ) {
        Text(text, style = VelaTypography.label(16f))
    }
}

// MARK: - Accent Button (matches iOS VelaAccentButtonStyle)

@Composable
fun VelaAccentButton(
    text: String,
    onClick: () -> Unit,
    modifier: Modifier = Modifier,
    enabled: Boolean = true,
    isLoading: Boolean = false,
) {
    Button(
        onClick = onClick,
        modifier = modifier.fillMaxWidth().height(54.dp),
        enabled = enabled && !isLoading,
        shape = VelaButtonShape,
        colors = ButtonDefaults.buttonColors(
            containerColor = VelaColor.accent,
            contentColor = Color.White,
            disabledContainerColor = VelaColor.accent.copy(alpha = 0.5f),
            disabledContentColor = Color.White.copy(alpha = 0.7f),
        ),
    ) {
        if (isLoading) {
            CircularProgressIndicator(
                modifier = Modifier.size(20.dp),
                color = Color.White,
                strokeWidth = 2.dp,
            )
        } else {
            Text(text, style = VelaTypography.label(16f))
        }
    }
}

// MARK: - Sail Logo (matches iOS VelaSailLogo)

@Composable
fun VelaSailLogo(
    size: Dp = 80.dp,
    color: Color = VelaColor.accent,
) {
    Canvas(modifier = Modifier.size(size)) {
        val w = this.size.width
        val h = this.size.height

        // Sail shape
        val sail = Path().apply {
            moveTo(w * 0.475f, h * 0.15f)
            quadraticTo(w * 0.3f, h * 0.5f, w * 0.225f, h * 0.8f)
            quadraticTo(w * 0.4f, h * 0.72f, w * 0.5f, h * 0.675f)
            quadraticTo(w * 0.6f, h * 0.72f, w * 0.775f, h * 0.8f)
            quadraticTo(w * 0.7f, h * 0.5f, w * 0.525f, h * 0.15f)
            close()
        }
        drawPath(sail, color, style = Fill)

        // Mast
        drawLine(
            color = color,
            start = Offset(w * 0.5f, h * 0.675f),
            end = Offset(w * 0.5f, h * 0.9f),
            strokeWidth = 2.dp.toPx(),
        )

        // Base line
        drawLine(
            color = color.copy(alpha = 0.4f),
            start = Offset(w * 0.375f, h * 0.9f),
            end = Offset(w * 0.625f, h * 0.9f),
            strokeWidth = 1.5.dp.toPx(),
        )
    }
}

// MARK: - Network Dot

@Composable
fun NetworkDot(
    color: Color = VelaColor.green,
    size: Dp = 7.dp,
) {
    Box(
        modifier = Modifier
            .size(size)
            .background(color, CircleShape),
    )
}

// MARK: - Token Icon

@Composable
fun TokenIcon(
    label: String,
    color: Color,
    bg: Color,
    size: Dp = 42.dp,
) {
    Box(
        modifier = Modifier
            .size(size)
            .background(bg, CircleShape),
        contentAlignment = Alignment.Center,
    ) {
        Text(
            text = label,
            style = VelaTypography.label(size.value * 0.38f),
            color = color,
        )
    }
}

// MARK: - Card Modifier

@Composable
fun VelaCard(
    modifier: Modifier = Modifier,
    content: @Composable ColumnScope.() -> Unit,
) {
    Column(
        modifier = modifier
            .background(VelaColor.bgCard, VelaCardShape)
            .border(1.dp, VelaColor.border, VelaCardShape),
        content = content,
    )
}

File: ios/VelaWallet/VelaWallet/Views/Components/VelaComponents.swift
import SwiftUI

// MARK: - Navigation Bar

struct VelaNavBar: View {
    let title: LocalizedStringResource
    var onBack: (() -> Void)?

    var body: some View {
        HStack {
            if let onBack {
                Button(action: onBack) {
                    Image(systemName: "chevron.left")
                        .font(.system(size: 17, weight: .semibold))
                        .foregroundStyle(VelaColor.textPrimary)
                        .frame(width: 36, height: 36)
                }
            } else {
                Spacer().frame(width: 36)
            }

            Spacer()

            Text(title)
                .font(VelaFont.title(17))
                .foregroundStyle(VelaColor.textPrimary)

            Spacer()
            Spacer().frame(width: 36)
        }
        .padding(.horizontal, VelaSpacing.screenH)
        .padding(.vertical, 8)
    }
}

// MARK: - Section Header

struct SectionHeader: View {
    let title: LocalizedStringResource

    var body: some View {
        Text(title)
            .font(.system(size: 11, weight: .semibold))
            .tracking(1.5)
            .foregroundStyle(VelaColor.textTertiary)
            .frame(maxWidth: .infinity, alignment: .leading)
            .padding(.horizontal, VelaSpacing.screenH)
    }
}

// MARK: - Network Dot

struct NetworkDot: View {
    var color: Color = VelaColor.green
    var size: CGFloat = 7

    var body: some View {
        Circle()
            .fill(color)
            .frame(width: size, height: size)
    }
}

// MARK: - Token Icon

struct TokenIcon: View {
    let label: String
    let color: Color
    let bg: Color
    var size: CGFloat = 42

    var body: some View {
        ZStack {
            Circle()
                .fill(bg)
            Text(label)
                .font(VelaFont.label(size * 0.38))
                .foregroundStyle(color)
        }
        .frame(width: size, height: size)
    }
}

// MARK: - Network Icon

struct NetworkIcon: View {
    let network: Network
    var size: CGFloat = 40

    var body: some View {
        ZStack {
            RoundedRectangle(cornerRadius: size * 0.3)
                .fill(network.iconBg)
            Text(network.iconLabel)
                .font(VelaFont.label(size * 0.35))
                .foregroundStyle(network.iconColor)
        }
        .frame(width: size, height: size)
    }
}

// MARK: - Sail Logo

struct VelaSailLogo: View {
    var size: CGFloat = 80
    var color: Color = VelaColor.accent

    var body: some View {
        Canvas { context, canvasSize in
            let w = canvasSize.width
            let h = canvasSize.height

            // Sail shape
            var sail = Path()
            sail.move(to: CGPoint(x: w * 0.475, y: h * 0.15))
            sail.addQuadCurve(
                to: CGPoint(x: w * 0.225, y: h * 0.8),
                control: CGPoint(x: w * 0.3, y: h * 0.5)
            )
            sail.addQuadCurve(
                to: CGPoint(x: w * 0.5, y: h * 0.675),
                control: CGPoint(x: w * 0.4, y: h * 0.72)
            )
            sail.addQuadCurve(
                to: CGPoint(x: w * 0.775, y: h * 0.8),
                control: CGPoint(x: w * 0.6, y: h * 0.72)
            )
            sail.addQuadCurve(
                to: CGPoint(x: w * 0.525, y: h * 0.15),
                control: CGPoint(x: w * 0.7, y: h * 0.5)
            )
            sail.closeSubpath()
            context.fill(sail, with: .color(color))

            // Mast
            var mast = Path()
            mast.move(to: CGPoint(x: w * 0.5, y: h * 0.675))
            mast.addLine(to: CGPoint(x: w * 0.5, y: h * 0.9))
            context.stroke(mast, with: .color(color), lineWidth: 2)

            // Base line
            var base = Path()
            base.move(to: CGPoint(x: w * 0.375, y: h * 0.9))
            base.addLine(to: CGPoint(x: w * 0.625, y: h * 0.9))
            context.stroke(base, with: .color(color.opacity(0.4)), lineWidth: 1.5)
        }
        .frame(width: size, height: size)
    }
}
